scripts/bump-version.js

```diff
@@ -0,0 +1,215 @@
+      rc = null;
+      break;
+    case 'rc':
+      if (rc) {
+        rc += 1;
+      } else {
+        rc = 1;
+      }
+      break;
+    case 'release':
+      rc = null;
+      break;
+    default:
+      throw new Error(`Invalid update type: ${_updateType}`);
+  }
+
+  const mainVersionString = `${major}.${minor}.${patch}`;
+  const rcVersion = rc;
+  const updatedVersionString = `${mainVersionString}${rc ? `-rc${rc}` : ''}`;
+  return {
+    mainVersionString,
+    rcVersion,
+    updatedVersionString,
+  }
+}
+
+/**
+ * Updates the version number on package.json and returns an object containing the updated values
+ * to be used on other file updates
+ * @param {UpdateType} _updateType
+ * @returns {{rcVersion: number, updatedVersionString: string, mainVersionString: string}}
+ */
+function updatePackageJson(_updateType) {
+  const packageJsonPath = path.join(__dirname, '..', 'package.json');
+  const packageJsonFile = fs.readFileSync(packageJsonPath, 'utf8');
+
+  // Look for the line containing "version"
+  const versionNameRegex = /"version": "(.+)"/;
+  const versionNameMatch = packageJsonFile.match(versionNameRegex);
+  const currentVersionString = versionNameMatch[1];
+  const updatedVersionObj = calculateNewVersion(_updateType, currentVersionString);
+
+  // Replace the version code on the file contents
+  const updatedPackageJsonFile = packageJsonFile.replace(versionNameRegex, `"version": "${updatedVersionObj.updatedVersionString}"`);
+
+  // Replace the file on disk
+  fs.writeFileSync(packageJsonPath, updatedPackageJsonFile);
+
+  // Return the updated version numbers in an object
+  return updatedVersionObj;
+}
+
+function updatePackageLockJson(updatedVersionString) {
+  const packageLockJsonPath = path.join(__dirname, '..', 'package-lock.json');
+  const packageLockJsonFile = fs.readFileSync(packageLockJsonPath, 'utf8');
+
+  /*
+   * This file contains versions for many different dependencies, so a simple regex approach won't
+   * work. We'll have to parse the file line by line, identify the code blocks related to the
+   * mobile app and update only their version numbers.
+   */
+  const lines = packageLockJsonFile.split('\n');
+  let nextVersionLine = -1;
+  let updatedLinesCount = 0;
+  for (let i = 0; i < lines.length; i += 1) {
+    if (updatedLinesCount === 2) {
+      break; // No need to continue: all the needed versions were already updated
+    }
+
+    const line = lines[i];
+    if (line.includes('"name": "HathorMobile"')) {
+      // Found a block related to this app, will update its version number
+      nextVersionLine = i + 1;
+      continue;
+    }
+    if (i === nextVersionLine) {
+      // Replace the app version on this line
+      const versionNameRegex = /"version": ".+"/;
+      lines[i] = line.replace(versionNameRegex, `"version": "${updatedVersionString}"`);
+      updatedLinesCount += 1;
+    }
+  }
+
+  // Update the file contents
+  fs.writeFileSync(packageLockJsonPath, lines.join('\n'));
+}
```

Comment:
[suggestion 💬 ] Why not collect the version lines first, and then run a line replace? I believe it would be simpler and more readable. Each step with its own specialty.

Reply:
Thanks! Please check if it has become clearer on [6a1f645](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/383/commits/6a1f6450edcb2725268fb08a314928f932ff548e).